src/TransitionGroup.js

```diff
@@ -87,9 +87,8 @@ class TransitionGroup extends React.Component {
   }
 
   performAppear = (key) => {
-    this.currentlyTransitioningKeys[key] = true;
-
     let component = this.childRefs[key];
+    this.currentlyTransitioningKeys[key] = component;
```

Comment:
my concern with this approach is that there is now a worry that refs will be kept around after they are unmounted accidentally. If any case is missed you'll end up with a hash of objects that are being kept alive forever